permissions_validators/src/public_blockchain/transfer.rs

```diff
@@ -103,3 +115,133 @@ impl<W: WorldTrait> IsGrantAllowed<W> for GrantMyAssetAccess {
         check_asset_owner_for_token(&permission_token, authority)
     }
 }
+
+/// Validator that checks that `Transfer` instruction execution count
+/// fits well in some time period
+#[derive(Debug, Clone, Copy)]
+pub struct ExecutionCountFitsInLimit;
+
+impl_from_item_for_instruction_validator_box!(ExecutionCountFitsInLimit);
+
+impl<W: WorldTrait> IsAllowed<W, Instruction> for ExecutionCountFitsInLimit {
+    fn check(
+        &self,
+        authority: &AccountId,
+        instruction: &Instruction,
+        wsv: &WorldStateView<W>,
+    ) -> Result<(), DenialReason> {
+        if !matches!(instruction, Instruction::Transfer(_)) {
+            return Ok(());
+        };
+
+        let params = match retrieve_permission_params(wsv, authority)? {
+            Some(params) => params,
+            None => return Ok(()),
+        };
+
+        let period = retrieve_period(&params)?;
+        let count = retrieve_count(&params)?;
+
+        if count_executions(wsv, authority, period) >= count {
+            return Err(DenialReason::from(
+                "Transfer transaction limit for current period is exceed",
+            ));
+        }
+        Ok(())
+    }
+}
+
+/// Retrieve permission parameters for `ExecutionCountFitsInLimit` validator.
+/// Returns Some if there is a suitable token and None if not
+///
+/// # Errors
+/// - Account doesn't exist
+fn retrieve_permission_params<W: WorldTrait>(
+    wsv: &WorldStateView<W>,
+    authority: &AccountId,
+) -> Result<Option<BTreeMap<Name, Value>>, DenialReason> {
+    wsv.map_account(authority, |account| {
+        wsv.account_permission_tokens(account)
+            .iter()
+            .filter(|token| token.name == *CAN_TRANSFER_ONLY_FIXED_NUMBER_OF_TIMES_PER_PERIOD)
+            .map(|token| token.params.clone())
+            .next()
+    })
+    .map_err(|e| e.to_string())
+}
+
+/// Retrieve period from `params`
+///
+/// # Errors
+/// - There is no period parameter
+/// - Period has wrong value type
+fn retrieve_period(params: &BTreeMap<Name, Value>) -> Result<Duration, DenialReason> {
+    match params
+        .get(&*PERIOD_PARAM_NAME)
+        .ok_or_else(|| format!("Expected `{}` parameter", *PERIOD_PARAM_NAME))?
+    {
+        Value::U128(period) => Ok(Duration::from_millis(
+            u64::try_from(*period).map_err(|e| e.to_string())?,
+        )),
+        _ => Err(format!(
+            "`{}` parameter has wrong value type. Expected `u128`",
```

Comment:
we won't allow specifying period in other integer types?

Reply:
and what's the point of using `u128` if we're going to convert to `u64`?

Reply:
also, if this is going to remain like this, I think `Error` section in docs should include integer conversion error

Reply:
> we won't allow specifying period in other integer types?

Yeah. I think other integer types won't add more flexebility and it would be just two different ways to achieve the same result and I'm not a fan of it

> and what's the point of using `u128` if we're going to convert to `u64`?

The reason is that we don't have `u64` in `Value` and I'm not sure if we should add it just for this test case.
Another reason is that all milliseconds in Iroha 2 have `u128` type and we often convert them to `u64` in order to pass it inside `Duration`.
I think initialy the problem is [here](https://github.com/rust-lang/rust/issues/58580)